src/main/kotlin/com/atlassian/performance/tools/report/api/judge/RelativeNonparametricPerformanceJudge.kt

```diff
@@ -1,20 +1,38 @@
 package com.atlassian.performance.tools.report.api.judge
 
+import com.atlassian.performance.tools.jiraactions.api.ActionMetric
 import com.atlassian.performance.tools.jiraactions.api.ActionType
 import com.atlassian.performance.tools.report.ActionMetricsReader
-import com.atlassian.performance.tools.report.api.ShiftedDistributionRegressionTest
 import com.atlassian.performance.tools.report.api.junit.FailedAssertionJUnitReport
 import com.atlassian.performance.tools.report.api.junit.JUnitReport
 import com.atlassian.performance.tools.report.api.junit.SuccessfulJUnitReport
+import com.atlassian.performance.tools.report.api.result.DurationData
 import com.atlassian.performance.tools.report.api.result.EdibleResult
 import com.atlassian.performance.tools.report.junit.FailedActionJunitReport
 import com.atlassian.performance.tools.report.toPercentage
 
-class RelativeNonparametricPerformanceJudge(
-    private val significance: Double
+class RelativeNonparametricPerformanceJudge private constructor(
+    private val significance: Double,
+    private val actionMetricsReader: (metrics: List<ActionMetric>) -> Map<String, DurationData>,
+    private val shiftedDistributionRegressionTestProvider: ShiftedDistributionRegressionTestProvider
 ) {
 
-    constructor() : this(significance = 0.05)
+    class Builder {
```

Comment:
nit: The convention we are using everywhere in the modules connected to JPT is that builder child classes are at the end of the class definition. We could change this convention, however I'd suggest to keep it consistant.

Reply:
Indeed, [official Kotlin code style](https://kotlinlang.org/docs/coding-conventions.html#class-layout) says:

> Put nested classes next to the code that uses those classes. If the classes are intended to be used externally and aren't referenced inside the class, put them in the end, after the companion object.